userspace/libsinsp/sinsp.h

```diff
@@ -1223,17 +1223,30 @@ VISIBILITY_PRIVATE
 	using sinsp_evt_ptr = std::unique_ptr<sinsp_evt>;
 	struct state_evts_less
 	{
-		bool operator()(const sinsp_evt_ptr& l, const sinsp_evt_ptr& r)
+		bool operator()(const sinsp_evt& l, const sinsp_evt& r)
```

Comment:
Since the queue enforces using shared_ptr or unique_ptr, I thought it was a bit too verbose to re-define all methods and functons to accept std::unique_ptr<X> instead of just X. Happy to turn it back if you prefer!

Reply:
got it, thank you it's ok like it is!